### PR TITLE
[IMP][14.0] operating unit: updated validation for allowed operating unit

### PR DESCRIPTION
**Pull request by AshishHirapara**
Updated restriction on Selection of default operating unit based on the selected operating units on the allowed list

### CODE REVIEW

File: operating_unit/models/res_users.py [hunk 1]
```diff
@@ -44,6 +46,21 @@ def _default_operating_units(self):
...
+    def write(self, vals):
+        res = super(ResUsers, self).write(vals)
+        if any(
+            user.default_operating_unit_id
+            and user.default_operating_unit_id not in user.operating_unit_ids
+            for user in self
+        ):
+            raise ValidationError(
+                _(
+                    """The chosen Operating Unit is not in
```
Comment (AaronHForgeFlow) [inline operating_unit/models/res_users.py]: please use single quotes, otherwise the message does not display correctly

**Review by AaronHForgeFlow — COMMENTED** [review]
I think you can add a domain in the view so only the ones that are allowed are selectable.

File: operating_unit/models/res_users.py (hunk 1 above)
Comment (AshishHirapara) [reply operating_unit/models/res_users.py]: I have removed this code and made the validation through XML itself.

File: operating_unit/__manifest__.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
     "name": "Operating Unit",
     "summary": "An operating unit (OU) is an organizational entity part of a "
     "company",
-    "version": "14.0.1.0.2",
+    "version": "14.0.1.0.3",
```
Comment (francesco-ooops) [inline operating_unit/__manifest__.py]: @AshishHirapara do not update

**Comment by francesco-ooops** [comment]
@AaronHForgeFlow what do you say? 

I can't test manually as runboat gives me a 503 error

**Review by francesco-ooops — CHANGES_REQUESTED** [review]
runbot is up, functional review: 

Cannot remove an OU from Mitchell Admin: https://recordit.co/7kXaE9580d

**Comment by AshishHirapara** [comment]
> 

@AaronHForgeFlow  Regarding @francesco-ooops comment, 
This line https://github.com/AshishHirapara/operating-unit/blob/2865eb75bab756250b3f5040f737f58c27f9bcf2/operating_unit/models/res_users.py#L72 has a code to assign all the OUs in the system if the role is manager, so even if someone removes any OU from that user, this line will again assign all the OUs to that user, so it appears that the OU is not getting removed. And I think that line is required also in the flow, so what do you suggest to get work around in this case?

**Comment by AaronHForgeFlow** [comment]
Hi @AshishHirapara , in that case maybe we should go back to the previous approach using the validation error. Sorry for the inconvenience. When I commented that, it was just a suggestion that could be discussed, I did not mean you to change it right away, but thank you anyway for attending the comments.

**Comment by AshishHirapara** [comment]
Hi, @AaronHForgeFlow thanks for letting me know, I would like to revert the changes. Initially, I thought that was the right way to implement it, but I guess it's not. I will update the changes.

**Comment by AshishHirapara** [comment]
@francesco-ooops @AaronHForgeFlow it came to my notice that even after using the validation error approach, the issue will not be resolved. And the OUs will be assigned anyway to the manager role user.

**Review by francesco-ooops — APPROVED** [review]
@AaronHForgeFlow @AshishHirapara on second analysis, I think we can leave it like this. It's correct that user with group "OU Manager" has all OUs assigned and it's up to who is managing access rights to know that OUs cannot be removed from user "OU manager".

Functional review ok

**Review by SicurSam — APPROVED** [review]
functional ok!

**Comment by francesco-ooops** [comment]
@AaronHForgeFlow what do you say? :)

File: operating_unit/__manifest__.py (hunk 2 above)
Comment (AaronHForgeFlow) [reply operating_unit/__manifest__.py]: please do not update this

File: operating_unit/tests/test_operating_unit.py [hunk 3]
```diff
@@ -120,3 +121,21 @@ def test_02_operating_unit(self):
...
             user.default_operating_unit_id, default_user.default_operating_unit_id
         )
+        nou = self.env["operating.unit"].search(
+            [
+                "|",
+                ("company_id", "=", False),
+                ("company_id", "in", self.user1.company_ids.ids),
+            ]
+        )
+        # user_form = Form(self.env["res.users"], view='base.view_users_form')
```
Comment (AaronHForgeFlow) [inline operating_unit/tests/test_operating_unit.py]: please remove comments

File: operating_unit/tests/test_operating_unit.py [hunk 4]
```diff
@@ -120,3 +121,21 @@ def test_02_operating_unit(self):
...
+        partner = self.env["res.partner"].search([], limit=1)
+        with Form(self.env["res.users"], view="base.view_users_form") as user_form:
+            # user_form.operating_unit_ids.clear()
```
Comment (AaronHForgeFlow) [inline operating_unit/tests/test_operating_unit.py]: this one also

**Review by AaronHForgeFlow — COMMENTED** [review]
LGTM. Minor comments

File: operating_unit/tests/test_operating_unit.py (hunk 4 above)
Comment (AshishHirapara) [reply operating_unit/tests/test_operating_unit.py]: done

File: operating_unit/tests/test_operating_unit.py (hunk 3 above)
Comment (AshishHirapara) [reply operating_unit/tests/test_operating_unit.py]: done

**Comment by francesco-ooops** [comment]
@AaronHForgeFlow merge? :)

**Comment by AshishHirapara** [comment]
> LGTM. Minor comments

@AaronHForgeFlow all the changes you requested are done!

**Review by AaronHForgeFlow — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
/ocabot merge patch

@AshishHirapara  Can you please forward port it to 15.0?

**Comment by francesco-ooops** [comment]
@AaronHForgeFlow #511 :)

**Pull request merged**